src/Symfony/Bundle/FrameworkBundle/Routing/Router.php

```diff
@@ -172,18 +172,25 @@ private function resolve($value)
                 return '%%';
             }
 
+            if (isset($this->collectedParameters[$match[1]])) {
+                return $this->collectedParameters[$match[1]];
+            }
+
             if (preg_match('/^env\((?:\w++:)*+\w++\)$/', $match[1])) {
                 throw new RuntimeException(sprintf('Using "%%%s%%" is not allowed in routing configuration.', $match[1]));
             }
 
             $resolved = ($this->paramFetcher)($match[1]);
 
             if (\is_bool($resolved)) {
+                $this->collectedParameters[$match[1]] = $resolved;
                 $resolved = (string) (int) $resolved;
             }
 
             if (\is_string($resolved) || is_numeric($resolved)) {
-                $this->collectedParameters[$match[1]] = $resolved;
+                if (!isset($this->collectedParameters[$match[1]])) {
```

Comment:
Got it. Instead of this check, shouldn't we turn the previous "if" into an "elseif"?

Reply:
@nicolas-grekas We don't have `$resolved` at that point yet. We could move the entire if statement after `$resolved = ($this->paramFetcher)($match[1]);` but that would partially defeat the purpose as the first if statement should prevent any unnecessary code execution.

Reply:
Another possibility that would work:

```php
            if (\is_string($resolved) || is_numeric($resolved) || \is_bool($resolved)) {
                $this->collectedParameters[$match[1]] = $resolved;

                if (\is_bool($resolved)) {
                    $resolved = (string) (int) $resolved;
                }

                return (string) $this->resolve($resolved);
            }
```

Reply:
I updated the code, thanks :)